File: roveagent/workforce/personas.py
"""RoveFrame Executive Layer personas (single RoveAgent runtime).

五个角色共享同一 runtime，仅以 persona/skills/permissions/workflows
区分：CEO Insight / COO / CMO / CTO(system health) / Developer(研发)。

Developer 是后补的一个**真实能力黑洞**：运行时 `api/capability_router.py` 里
`developer` 一直存在且工具集最全（file/terminal/git/skills），但前端只暴露了
前四个 persona，没有任何一个映射到它 —— 用户在界面上永远选不到会写代码的角色，
只会得到 CEO Agent 的一句 "I don't have access to a terminal or file system"。
"""

from __future__ import annotations

PERSONAS: dict[str, dict[str, object]] = {
    "ceo-insight": {
        "name": "CEO Insight Agent",
        "label": "CEO Insight",
        "employee_key": "ceo",
        "mission": (
            "基于全局经营数据给业主提供 business overview 与战略建议；"
            "识别经营风险并推动解决；对业主负责。"
        ),
        "focus": "business overview + strategy",
    },
    "coo": {
        "name": "COO Agent",
        "label": "COO",
        "employee_key": "operations",
        "mission": (
            "负责日常运营：订单、库存、预约、差评与流程效率；"
            "发现运营异常并提出可执行的优化方案。"
        ),
        "focus": "operations",
    },
    "cmo": {
        "name": "CMO Agent",
        "label": "CMO",
        "employee_key": "marketing",
        "mission": (
            "负责客户增长与留存：分析流失高价值客户、起草召回活动、"
            "内容与活动复盘；外发动作必须提交业主审批。"
        ),
        "focus": "customer growth",
    },
    "cto": {
        "name": "CTO Agent",
        "label": "CTO",
        "employee_key": "devops",
        "mission": (
            "负责系统健康：监控部署、连接器同步、通知投递与自愈闭环；"
            "暂不承担产品研发职责。"
        ),
        "focus": "system health",
    },
    "developer": {
        "name": "Developer Agent",
        "label": "Developer",
        "employee_key": "developer",
        "mission": (
            "负责产品研发：读写代码、打补丁、跑测试、提交变更；"
            "写文件与终端命令需业主审批后执行。"
        ),
        "focus": "code, patch, test & ship",
    },
}

# persona key -> employee key（兼容旧 employee key 直接传）
PERSONA_ALIASES: dict[str, str] = {
    "ceo": "ceo",
    "ceo-insight": "ceo",
    "operations": "operations",
    "coo": "operations",
    "marketing": "marketing",
    "cmo": "marketing",
    "devops": "devops",
    "cto": "devops",
    "developer": "developer",
}


def persona_for(agent_key: str) -> dict[str, object] | None:
    """Return persona metadata for a persona key, or None. """
    return PERSONAS.get(agent_key)

File: src/lib/agent/personas.ts
/**
 * RoveFrame Executive Layer personas —— 统一 RoveAgent Runtime，
 * 以 persona/skills/permissions/workflows 区分（不新建 runtime）。
 *
 * `employeeKey` 必须对应运行时 `roveagent/api/capability_router.py` 里
 * AGENT_CAPABILITIES 的一个 key —— 否则界面选得中、运行时给不出对应工具集。
 * 这条一致性由 tests/persona-runtime-parity.test.ts 锁定。
 */
export type PersonaKey = 'ceo-insight' | 'coo' | 'cmo' | 'cto' | 'developer';

export interface Persona {
  key: PersonaKey;
  label: string;
  description: string;
  employeeKey: string;
}

export const PERSONAS: Persona[] = [
  {
    key: 'ceo-insight',
    label: 'CEO Insight',
    description: 'Business overview & strategy',
    employeeKey: 'ceo',
  },
  {
    key: 'coo',
    label: 'COO',
    description: 'Operations',
    employeeKey: 'operations',
  },
  {
    key: 'cmo',
    label: 'CMO',
    description: 'Customer growth',
    employeeKey: 'marketing',
  },
  {
    key: 'cto',
    label: 'CTO',
    description: 'System health',
    employeeKey: 'devops',
  },
  /**
   * 研发 Agent。**这个入口此前缺失，是一个真实的能力黑洞。**
   *
   * 运行时里 `developer` 一直存在且工具集最全：
   *   toolsets=("file","terminal","todo","git","skills","delegation")
   *   summary="AI 软件工程师：读写代码、打补丁、跑测试、git 提交（写与提交需审批）"
   * 但前端只有 4 个 persona（ceo / operations / marketing / devops），**没有任何一个
   * 映射到 `developer`** —— 这个编码 agent 从界面上永远够不到。
   *
   * 用户反馈「这个不能编码啊」时，答话的是 CEO Agent，原话
   * "I don't have access to a terminal or file system"。而 CEO 的能力画像里
   * 本就写着「不碰文件与终端」—— 它没说错，是**用户根本没有可选的编码角色**。
   */
  {
    key: 'developer',
    label: 'Developer',
    description: 'Code, patch, test & ship',
    employeeKey: 'developer',
  },
];

const PERSONA_KEYS: ReadonlySet<string> = new Set(PERSONAS.map((p) => p.key));

/** 校验并归一化客户端传入的 persona key（缺省 CEO Insight）。 */
export function resolvePersonaKey(value: unknown): PersonaKey {
  if (typeof value === 'string' && PERSONA_KEYS.has(value)) return value as PersonaKey;
  return 'ceo-insight';
}
